### PR TITLE
Lender Manager

### PR DESCRIPTION
**Pull request by sypeer**
TLRV2-124
TLRV2-126
TLRV2-255
TLRV2-416
TLRV2-417

### CODE REVIEW

File: packages/contracts/contracts/TellerV2Storage.sol [hunk 1]
```diff
@@ -41,7 +42,7 @@ struct Payment {
 struct Bid {
     address borrower;
     address receiver;
-    address lender;
+    address _lender; // DEPRECATED
```
Comment (ethereumdegen) [inline packages/contracts/contracts/TellerV2Storage.sol:45]: I personally dont love the idea that 'lender' will no longer be stored in the Storage contract.  I think that it would be the most clean if we kept this in here and just allowed the LenderManager to modify this storage.

Comment (ethereumdegen) [reply packages/contracts/contracts/TellerV2Storage.sol:45]: However if we do end up storing the 'lender' inside of the lender manager storage, we should really consider structuring it more like a NFT. Now is the time to do it.  We should make it follow the ERC721 protocol so that loans can be transferred like an NFT optionally in the future...!!!

File: packages/contracts/contracts/TellerV2.sol [hunk 2]
```diff
@@ -465,11 +494,8 @@ contract TellerV2 is
         Bid storage bid = bids[_bidId];
 
         address sender = _msgSenderForMarket(bid.marketplaceId);
-        (bool isVerified, ) = marketRegistry.isVerifiedLender(
```
Comment (ethereumdegen) [inline packages/contracts/contracts/TellerV2.sol:468]: are we sure that it is OK to remove this ?  It seems like now anyone would be able to accept bids now -- even if not verified by the market for that market

File: packages/contracts/contracts/TellerV2.sol [hunk 3]
```diff
@@ -945,7 +969,7 @@ contract TellerV2 is
         view
         returns (address lender_)
     {
-        lender_ = bids[_bidId].lender;
+        lender_ = bids[_bidId]._lender;
```
Comment (ethereumdegen) [inline packages/contracts/contracts/TellerV2.sol:972]: isnt this deprecated... ?

Comment (ethereumdegen) [reply packages/contracts/contracts/TellerV2.sol:972]: prob needs to call the lendermanager

File: packages/contracts/contracts/mock/TellerV2SolMock.sol [hunk 4]
```diff
@@ -161,7 +165,7 @@ contract TellerV2SolMock is ITellerV2, TellerV2Storage {
         view
         returns (address lender_)
     {
-        lender_ = bids[_bidId].lender;
+        lender_ = bids[_bidId]._lender;
```
Comment (ethereumdegen) [inline packages/contracts/contracts/mock/TellerV2SolMock.sol:168]: this is deprecated so needs to call lender manager

File: packages/contracts/contracts/TellerV2.sol (hunk 2 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/TellerV2.sol:468]: nevermind i see that it is handled inside of lender manager now

File: packages/contracts/contracts/LenderManager.sol [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+     * @param _newLender The address of the new active lender.
+     * @param _marketId The Id of the corresponding market.
+     */
+    function setNewLender(uint256 _bidId, address _newLender, uint256 _marketId)
+    public
+    override
+    {
+        address currentLender = _getActiveLender(_bidId);
+        if (currentLender == address(0)) {
+            _checkOwner();
```
Comment (ethereumdegen) [inline packages/contracts/contracts/LenderManager.sol:54]: why is this necessary

Comment (ethereumdegen) [reply packages/contracts/contracts/LenderManager.sol:54]: not adequately commented

**Comment by passabilities** [comment]
using #32 instead
